main.tf

```diff
@@ -34,7 +34,22 @@ resource "aws_iam_role" "ottertune_role" {
 
 data "aws_iam_policy_document" "ottertune_db_policy" {
   statement {
-    actions = [
+    actions = var.permissions_level == "write_limited" ? [
+      "budgets:Describe*",
+      "ce:Describe*",
+      "ce:Get*",
+      "ce:List*",
+      "cloudwatch:Describe*",
+      "cloudwatch:Get*",
+      "cloudwatch:List*",
+      "iam:SimulatePrincipalPolicy",
+      "pi:DescribeDimensionKeys",
+      "pi:GetResourceMetrics",
+      "rds:Describe*",
+      "rds:List*",
+      "rds:ModifyDBInstance",
+      "rds:ModifyDBCluster",
+    ] : [
```

Comment:
check out this answer, I think would look cleaner + more readable: https://stackoverflow.com/questions/67902785/adding-extra-element-to-a-list-in-terraform-if-condition-is-met